doc/src/manual/functions.md

```diff
@@ -533,7 +533,7 @@ this example, `width` is certain to have the value `2`.
 
 ## Evaluation Scope of Default Values
 
-When optional and keyword argument default expressions are evaluated, only *previous* arguments are in
+All arguments passed to a function are evaluated in lexical left-to-right order. When optional and keyword argument default expressions are evaluated, only *previous* arguments are in
```

Comment:
Correct. I think this actually deserves its own section heading (plus it doesn't really belong under "scope"). I'd make a `## Argument evaluation order` heading, put that sentence under, and maybe add an example.

Reply:
I'm working on an example. Just came across this odd error message while testing some edge cases of the syntax:

```julia

julia> f(a, b=println("b default"), c=println("c default"); k=println("k default")) = a
f (generic function with 3 methods)

julia> f(println("a"); k=println("k"), println("b"))
a
k
b
ERROR: MethodError: no method matching start(::Void)
Closest candidates are:
  start(::SimpleVector) at essentials.jl:534
  start(::Base.MethodList) at reflection.jl:654
  start(::ExponentialBackOff) at error.jl:147
  ...
Stacktrace:
 [1] anonymous at ./<missing>:?
```

Reply:
#9972

Reply:
I added an example and moved the section right below the section on "Argument destructuring" since it felt like it flowed better.